refactor(simple vouchers): update to standards

jniles opened the pull request:
This commit updates the simple vouchers to the latest 2.X standards.
1. The page now uses `Notify` for handling success and errors.
2. Use `bhCurrencySelect` for currency selection - includes bugfixes.
3. End to end tests use the new FU.typeahead() method to configure the
   typeaheads on the page.
4. UI updates
   1. The bhBreadcrumb component now does translation automatically
      instead of having to $translate in controllers.
   2. The breadcrumb now reads `Finance` / `Simple Voucher` instead of
      `Bhima`.

Additionally, all `<textarea>`s in the entire application are limited to resizing vertically, and are limited to a maximum height of 250px.

---

Thank you for contributing!

Before submitting this pull request, please verify that you have:
- [x] Run your code through JSHint.  [Check out our styleguide](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/docs/STYLEGUIDE.md).
- [x] Run integration tests.
- [x] Run end-to-end tests.
- [x] Accurately described the changes your are making in this pull request.

For a more detailed checklist, [see the official review checklist](https://docs.google.com/document/d/1nupLVLRXgSZJQo_acLgrwvPnN8RukfSiwRhSToj81uU/pub) that this PR will be evaluated against.

Ensuring that the above checkboxes are completed will help speed the review process and help build a stronger application.  Thanks!

jniles committed:
refactor(simple vouchers): update to standards

This commit updates the simple vouchers to the latest 2.X standards.

 1. The page now uses `Notify` for handling success and errors.
 2. Use `bhCurrencySelect` for currency selection - includes bugfixes.
 3. End to end tests use the new FU.typeahead() method to configure the
 typeaheads on the page.
 4. UI updates
   1. The bhBreadcrumb component now does translation automatically
   instead of having to $translate in controllers.
   2. The breadcrumb now reads `Finance` / `Simple Voucher` instead of
   `Bhima`.

sfount commented on client/src/partials/vouchers/simple.js:
```diff
@@ -2,116 +2,73 @@ angular.module('bhima.controllers')
-    delete vm.httpError;
-
     // stop submission if the form is invalid
     if (form.$invalid) {
```
A number of pages use `Notify.danger` here to inform that the form is incomplete or incorrect. I'm not very familiar with the module, does it make sense in this case?

jniles replied:
It might make sense - I'll give it a try.

My original thinking was that the form is only five inputs long, so they should all be visible if validation occurs.  But if all other forms use `Notify.danger`, I'll conform to standards.

jniles committed:
fix: add error notification for $invalid form

This commit adds a Notify.danger() call to alert the user that an error
occurred in their form.
